### ISSUE TITLE
Upgrade to v4.2.6 fails: nsa commit not found

### ISSUE THREAD

**Issue by 17dec** [mastodon/mastodon#29203]
### Steps to reproduce the problem

```sh
git fetch && git checkout v4.2.6
bundle install
```

### Expected behaviour

Successfully fetch & install dependencies

### Actual behaviour

nsa gem fails to install

### Detailed description

```
Bundler 2.3.26 is running, but your lockfile was generated with 2.4.13. Installing Bundler 2.4.13 and restarting using that version.
Fetching gem metadata from https://rubygems.org/.
Fetching bundler 2.4.13
Installing bundler 2.4.13
Fetching gem metadata from https://rubygems.org/.........
Fetching https://github.com/stanhu/omniauth-cas.git
Fetching https://github.com/jhawthorn/nsa.git
fatal: Could not parse object 'e020fcc3a54d993ab45b7194d89ab720296c111b'.
Git error: command `git reset --hard e020fcc3a54d993ab45b7194d89ab720296c111b` in directory /home/rubbish/mastodon/.local-stuff/bundle/ruby/3.1.0/bundler/gems/nsa-e020fcc3a54d has
failed.
Revision e020fcc3a54d993ab45b7194d89ab720296c111b does not exist in the repository https://github.com/jhawthorn/nsa.git. Maybe you misspelled it?
If this error persists you could try removing the cache directory '/home/rubbish/mastodon/.local-stuff/bundle/ruby/3.1.0/bundler/gems/nsa-e020fcc3a54d'
```

Clearing all the caches does not solve the problem.

### Mastodon version

v4.2.6

### Technical details

- Ruby version: ruby 3.1.4p223 (2023-03-30 revision 957bb7cb81) [x86_64-linux]
- Node.js version: v20.11.0

**Comment by legrex** [comment]
Seems different, but after (no-docker)
`git fetch && git checkout v4.2.6
bundle install` 
and 
`bundle install`
`yarn install --frozen-lockfile`
and restarting processes, i'm still at 4.2.5. Got no errors in the process though.

**Comment by 17dec** [comment]
> To fix it for now, you should be able to change the `Gemfile` and replace `gem 'nsa', github: 'jhawthorn/nsa', ref: 'e020fcc3a54d993ab45b7194d89ab720296c111b'` with `gem 'nsa'`, then run `bundle install`

Yup this did the trick, thanks!

**Comment by renchap** [comment]
This has been fixed in 4.2.7
